### PR TITLE
implemented paused state check for resume call

### PR DESCRIPTION
**Pull request by ChrisMuller80**
Fixes DD-366

# Description of changes
Checks the response of the resume call to see if the workflow has been paused in Dataverse.

# How to test

- Create and publish dataset
- Check log for success: `vagrant ssh` `tail -f /var/opt/dans.knaw.nl/log/dd-pre-publish-workflow/dd-pre-publish-workflow.log`
- In `SetVaultMetadataTask.scala` add `notPausedError=true` on line 67.  Redeploy and publish again. Check log.
- Also, replace `dataverse.workflows().resume(invocationId)` on line 65 with fake response. Redeploy, publish and check the log.

# Related PRs 
(Add links)
* 

# Notify
@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.dd.prepub/Configuration.scala [hunk 1]
```diff
@@ -15,17 +15,19 @@
...
 import org.apache.commons.configuration.PropertiesConfiguration
 
+import java.net.URI
+
 case class Configuration(version: String,
                          serverPort: Int,
                          pidGeneratorBaseUrl: URI,
-                         dataverse: DataverseInstanceConfig
+                         dataverse: DataverseInstanceConfig,
+                         AwaitWorkflowPausedStateMaxNumberOfRetries: Int,
```
Comment (janvanmansum) [inline src/main/scala/nl.knaw.dans.dd.prepub/Configuration.scala]: variable names should be in camelCase

File: src/main/scala/nl.knaw.dans.dd.prepub/package.scala [hunk 2]
```diff
@@ -63,4 +63,6 @@ package object prepub {
   }
   ))
 
+  case class WorkflowNotPausedException(numberOfTimesTried: Int, waitTimeInMilliseconds: Int) extends RuntimeException(s"Workflow was not paused. Number of tries = $numberOfTimesTried, wait time between tries = $waitTimeInMilliseconds ms.")
```
Comment (janvanmansum) [inline src/main/scala/nl.knaw.dans.dd.prepub/package.scala]: Move to line 32, close to the other exception types.

File: src/main/scala/nl.knaw.dans.dd.prepub/SetVaultMetadataTask.scala [hunk 3]
```diff
@@ -16,32 +16,81 @@
...
+   * @return Success if the resume call succeeded, otherwise a Failure.
+   */
+  private def resumeWorkflow(dataverse: DataverseInstance, invocationId: String, maxNumberOfRetries: Int, timeBetweenRetries: Int): Try[Unit] = {
+    trace(maxNumberOfRetries, timeBetweenRetries)
+    var numberOfTimesTried = 0
+    var notPausedError = true
+
+    do {
+      val resumeResponse = dataverse.workflows().resume(invocationId)
+      notPausedError = checkResponseForPausedError(resumeResponse, invocationId).unsafeGetOrThrow
```
Comment (janvanmansum) [inline src/main/scala/nl.knaw.dans.dd.prepub/SetVaultMetadataTask.scala]: I'd go for invocationIdNotFound in the function and value names. It is more literally what the error from Dataverse means and also slightly less prone to missreading (not PausedError vs notPaused Error)

**Review by janvanmansum — CHANGES_REQUESTED** [review]
It looks OK, except for some i's that need dotting.

Also, please resolve the conflicts with master.

File: src/main/scala/nl.knaw.dans.dd.prepub/SetVaultMetadataTask.scala (hunk 3 above)
Comment (ChrisMuller80) [reply src/main/scala/nl.knaw.dans.dd.prepub/SetVaultMetadataTask.scala]: I also renamed the WorkflowNotPausedException to InvocationIdNotFoundException

**Review by janvanmansum — APPROVED** [review]

**Comment by janvanmansum** [comment]
Looks good. I tested it by setting a breakpoint in Dataverse just after the step is invoked (make sure that you set the breakpoint to only pause the thread!) and if I wait long a enough the resume-attempts fail. This proves that the loop is actually useful. Without the breakpoint the workflow was successfully resumed.

**Pull request merged**